perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_unixGID_namespace.java

```diff
@@ -94,26 +94,30 @@ public Attribute fillAttribute(PerunSessionImpl sess, Resource resource, Attribu
 	}
 
 	@Override
-	public void checkAttributeSemantics(PerunSessionImpl sess, Resource resource, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException{
+	public void checkAttributeSemantics(PerunSessionImpl sess, Resource resource, Attribute attribute) throws InternalErrorException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException{
 		try{
 			String gidNamespace = attribute.getFriendlyNameParameter();
 
 			//Special behaviour if gid is null
 			Integer attrValue;
 			if(attribute.getValue() == null) {
-				throw new WrongAttributeValueException(attribute, resource, "Unix GID must be set");
+				throw new WrongReferenceAttributeValueException(attribute, null, resource, null, "Unix GID must be set");
 			} else {
-				attrValue = (Integer) attribute.getValue();
+				attrValue = attribute.valueAsInteger();
 			}
 
 			//Check if GID is within allowed range
-			sess.getPerunBl().getModulesUtilsBl().checkIfGIDIsWithinRange(sess, attribute);
+			try {
+				sess.getPerunBl().getModulesUtilsBl().checkIfGIDIsWithinRange(sess, attribute);
```

Comment:
I think the method checkIfGIDIsWithinRange should be reworked, so it does not throw WrongAttributeValueException. This Exception is thrown on two places: first if the attribute is null where we usually do not throw Value exception. Second, from method checkAndConvertGIDRanges which check reference attribute so we should catch Value exception and throw the Reference exception.

Reply:
We already discussed this method earlier with group module and decided to this solution.